api/src/main/java/io/druid/indexer/TaskState.java

```diff
@@ -22,8 +22,11 @@
 public enum TaskState
 {
   RUNNING,
+  WAITING,
```

Comment:
Would you add descriptions for each state to here?

Reply:
And I wonder why these new states are needed in this PR. Would you share your thoughts more about this?

Reply:
This is an incompatible change. This PR should be tagged with `Release notes` and a note for rolling update should be added to the release notes if this change is needed.

Reply:
The `tasks` API returns all tasks if no `state` query param is passed, so the user of this API wants to know `statusCode` for each of the tasks. Current `statusCode` consist of `running`, `success` and `failed`. We need to also send appropriate `statusCode` for `waiting` and `pending`. That's why added those here. We can sure mark this PR as `Release notes` if you agree to this change.

Reply:
If I make another `RuntimeTaskState` do you suggest that to contain all the states : `running` `pending` `waiting` `complete `success` `failed` or just subclass of running i.e. `running` `waiting` `pending`. The reason I ask is `TaskStatusPlus` object contains a single property called `statusCode`, which is the `TaskState` and UI uses this property to fill status. Should we create two separate status codes or combine everything in one. I prefer latter, i.e. return one status code.

Reply:
I was thinking the runtime task status code should be its own thing (not a subclass of task state).

Reply:
okay, i added `RunnerTaskState` and now `TaskStatusPlus` will return 2 status codes for each task `statusCode` and `runnerStatusCode`.